libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2089,6 +2089,81 @@ MAV_RESULT GCS_MAVLINK::handle_command_preflight_set_sensor_offsets(const mavlin
     return MAV_RESULT_ACCEPTED;
 }
 
+bool GCS_MAVLINK::calibrate_gyros()
+{
+    AP::ins().init_gyro();
+    if (!AP::ins().gyro_calibrated_ok_all()) {
+        return false;
+    }
+    AP::ahrs().reset_gyro_drift();
+    return true;
+}
+
+MAV_RESULT GCS_MAVLINK::_handle_command_preflight_calibration_baro()
+{
+    gcs().send_text(MAV_SEVERITY_INFO, "Updating barometer calibration");
+    AP::baro().update_calibration();
+    gcs().send_text(MAV_SEVERITY_INFO, "Barometer calibration complete");
+
```

Comment:
Why not use the Airspeed singleton here and calibrate the airspeed sensor if it's not a nullptr? It would allow you to remove the virtual method/handling on baro from plane, as well as being ready to accept an airspeed sensor on any other vehicle if they choose to have them at a later point in time.